### PR TITLE
Update fork

### PR DESCRIPTION
**Pull request by mpenciak**
This PR:

* Renames the package to `FFC` from `FFaCiL`
  - The combination of upper and lower case, and the length of `FFaCiL` made it somewhat unpleasant to type
* Removes the dependence on Yatima Standard Library by migrating the necessary files to the `Util` submodule.
  - I don't plan on maintaining YSL which is a much larger codebase than what is needed for this library
  - It should be a top priority to minimize the `Util` files. I did a rough job of moving over the functionality from YSL, but a lot of it has since been superseded by std4 and a lot is likely not strictly necessary for functionality.
* Bumps the toolchain to `leanprover/lean4:nightly-2023-07-24`.

### CODE REVIEW

**Comment by mpenciak** [comment]
This still has a failing SWM test.

I believe the failing tests are coming from the change in Lean's integer division behavior which were discussed in the lean 4 dev Zulip channel...

There's also a totally possible change that I wasn't completely careful transferring the Yatima Standard Library files into `Util` which broke everything.

**Comment by mpenciak** [comment]
Tests pass now, the PR should be good for review!

File: FFC/EllipticCurve.lean [hunk 1]
```diff
@@ -267,8 +273,7 @@ Handbook of elliptic and hyperelliptic curve cryptography by Henri Cohen, et al.
 def double [Field F] {C : Curve F} :
   AffinePoint C → AffinePoint C
   | affine x y =>
-    let xx := x^2
-    let lambda := (xx + xx + xx + Curve.a C) / (y + y)
+    let lambda := ((3 : Nat) * x^2 + C.a) / ((2 : Nat) * y)
     let x' := lambda^2 - (2 : Nat) * x
```
Comment (DanielRrr) [inline FFC/EllipticCurve.lean:277]: Is that the reason for the bugs?

File: FFC/Util/AddChain.lean [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+## Yatima Standard Library
+
+Parts of this file were migrated from the corresponding file in the Yatima Standard Library which is
+no longer being maintained
+-/
+
+/-- The typeclass which implements efficient `add`, `mul`, `double`, and `doubleAdd` methods -/
+class Chainable (α : Type _) extends BEq α, OfNat α (nat_lit 1) where
+  add : α → α → α
```
Comment (DanielRrr) [inline FFC/Util/AddChain.lean:30]: Hm, this fairly looks like the `CurveGroup` class (or what we had)

File: README.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
-# FFaCiL.lean
+# FFC (Formerly FFaCiL)
 FFaCiL: Finite Fields and Curves in Lean,
 which is also "easy" in such languages as Portuguese and French ("fácil" and "facile" respectively).
```
Comment (DanielRrr) [inline README.md:3]: I would get rid of these lines if the name FFaCiL is no longer relevant

**Review by DanielRrr — APPROVED** [review]
Looks good to me, feel free to merge it

**Comment by DanielRrr** [comment]
> This still has a failing SWM test.
> 
> I believe the failing tests are coming from the change in Lean's integer division behavior which were discussed in the lean 4 dev Zulip channel...
> 
> There's also a totally possible change that I wasn't completely careful transferring the Yatima Standard Library files into `Util` which broke everything.

I would fix that as a separate issue.

**Pull request merged**